packages/opds-web-client/src/__tests__/DataFetcher-test.ts

```diff
@@ -107,21 +124,20 @@ describe("DataFetcher", () => {
   });
 
   describe("fetchOPDSData()", () => {
-    it("throws error if response isn't 200", (done) => {
+    it("throws error if response isn't 200", () => {
       mockFetch = stub().returns(new Promise<any>((resolve, reject) => {
         resolve({
           status: 401,
           text: () => new Promise((resolve, reject) => resolve("unauthorized"))
         });
       }));
-      fetch = mockFetch;
+      window.fetch = mockFetch;
 
       let fetcher = new DataFetcher();
-      fetcher.fetchOPDSData("test url").catch(err => {
+      fetcher.fetchOPDSData("test-url").catch(err => {
```

Comment:
I think you might have to return this for it to work? (looking at the mocha docs on returning a promise instead of using `done`)

Reply:
It times out when returning the promise because it seems to be expecting it to resolve and not reject.

Reply:
In that case I think you might have to use `done` - otherwise the test will return before the promise rejects. Can you make it fail if you change something inside the catch?

Reply:
Fixed the test and the function.